pkg/tecdsa/marshaling.go

```diff
@@ -0,0 +1,156 @@
+package tecdsa
+
+import (
+	"fmt"
+	"github.com/binance-chain/tss-lib/crypto"
+	"github.com/binance-chain/tss-lib/crypto/paillier"
+	"github.com/binance-chain/tss-lib/ecdsa/keygen"
+	"github.com/keep-network/keep-core/pkg/tecdsa/gen/pb"
+	"math/big"
+)
+
+// Marshal converts the PrivateKeyShare to a byte array.
+func (pks *PrivateKeyShare) Marshal() ([]byte, error) {
```

Comment:
In `Unmarshal` we assume the curve to be `tecdsa.Curve`. Should we validate the curve when marshaling? Just like noted in one of the previous comments, I think comparing the name should be enough. I see the curve is available in `LocalPartySaveData.ECDSAPub.Curve()`.

Reply:
See 659d9b61